frontend/src/pages/SignupPage/SignupPage.js

```diff
@@ -4,44 +4,141 @@ import Button from 'react-bootstrap/Button';
 import {useState} from 'react';
 import DropdownButton from 'react-bootstrap/DropdownButton';
 import Dropdown from 'react-bootstrap/Dropdown';
-
+import { useAtom, atom} from "jotai";
+import {currUser} from '../../atoms.js';
+import ErrorBar from '@/components/ErrorBar/ErrorBar';
+import {useRouter} from 'next/router'
 
 
 function SignupPage() {
+
+    const [user, setUser] = useAtom(currUser);
+
+    const [path, setPath] = useState();
+
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [gender, setGender] = useState('');
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const [graduationYear, setGraduationYear] = useState(0);
+    const [major, setMajor] = useState('');
+    
+    const [error, setError] = useState(null);
+
+    const router = useRouter();
+
+    const handleSubmit = async (e) => {
+        e.preventDefault(); 
+        setError(null);
+        const emptyFields = firstName == '' || lastName == '' || email == '' 
+            || password == '' || confirmPassword == '' || gender == '' 
+            || graduationYear == '' || major == '';
+
+        if (confirmPassword != password) {
+            setError("Make sure passwords match");
+            e.preventDefault();
+            return;
+        } else if (password.length < 8) {
+            setError("Password must have at least 8 characters");
+            e.preventDefault();
+            return;
+        }
+
+        if (emptyFields) {
+            setError("Please fill in all fields");
+            console.log("not null");
```

Comment:
let's try and remove all our console.log if it doesn't serve like a functional purpose